src/components/courses/InstructorSelector.tsx

```diff
@@ -19,10 +19,15 @@ export default function InstructorSelector({ instructors, setInstructors }: { in
 
     const instructorTags = useMemo(() => toTags(instructors), [instructors]);
 
+    function alreadyChosen(newInstructor: Instructor) {
+        return instructors.map(i => i.id).indexOf(newInstructor.id) !== -1;
+    }
+
     const addInstructor = ({ value }: { value: string }) => {
         console.log("addInstructor", value);
 
-        setInstructors([...instructors, JSON.parse(value)]);
+        let newInstructor = JSON.parse(value);
+        {!alreadyChosen(newInstructor) && setInstructors([...instructors, newInstructor])};
```

Comment:
The block statement and the usage of && here is very creative, I'd rather go with the more common `if(!isAlreadyChosen(newInstructor)) setInstructors(...);`